### ISSUE TITLE
JaCoCo dependency pulls many unnecessary Maven modules into users' own projects

### ISSUE THREAD

**Issue by shabino** [DanielYWoo/fast-object-pool#51]
This dependency in the POM causes many modules to be included:

```
<dependency>
      <groupId>org.jacoco</groupId>
      <artifactId>jacoco-maven-plugin</artifactId>
      <version>0.8.6</version>
      <type>maven-plugin</type>
</dependency>
```

I'm not sure why it's necessary as a "normal", non-plugin dependency.

Here's a snippet from my invocation of "mvn dependency:tree":

```
[INFO] +- cn.danielw:fast-object-pool:jar:2.2.0:compile
[INFO] |  \- org.jacoco:jacoco-maven-plugin:maven-plugin:0.8.6:compile
[INFO] |     +- org.apache.maven:maven-plugin-api:jar:3.0:compile
[INFO] |     |  +- org.apache.maven:maven-model:jar:3.0:compile
[INFO] |     |  +- org.apache.maven:maven-artifact:jar:3.0:compile
[INFO] |     |  \- org.sonatype.sisu:sisu-inject-plexus:jar:1.4.2:compile
[INFO] |     |     \- org.sonatype.sisu:sisu-inject-bean:jar:1.4.2:compile
[INFO] |     |        \- org.sonatype.sisu:sisu-guice:jar:noaop:2.1.7:compile
[INFO] |     +- org.apache.maven:maven-core:jar:3.0:compile
[INFO] |     |  +- org.apache.maven:maven-settings:jar:3.0:compile
[INFO] |     |  +- org.apache.maven:maven-settings-builder:jar:3.0:compile
[INFO] |     |  +- org.apache.maven:maven-repository-metadata:jar:3.0:compile
[INFO] |     |  +- org.apache.maven:maven-model-builder:jar:3.8.4:compile
[INFO] |     |  |  +- javax.inject:javax.inject:jar:1:compile
[INFO] |     |  |  +- org.apache.maven:maven-builder-support:jar:3.8.4:compile
[INFO] |     |  |  \- org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.5:compile
[INFO] |     |  +- org.apache.maven:maven-aether-provider:jar:3.0:runtime
[INFO] |     |  +- org.sonatype.aether:aether-impl:jar:1.7:compile
[INFO] |     |  |  \- org.sonatype.aether:aether-spi:jar:1.7:compile
[INFO] |     |  +- org.sonatype.aether:aether-api:jar:1.7:compile
[INFO] |     |  +- org.sonatype.aether:aether-util:jar:1.7:compile
[INFO] |     |  +- org.codehaus.plexus:plexus-interpolation:jar:1.14:compile
[INFO] |     |  +- org.codehaus.plexus:plexus-classworlds:jar:2.2.3:compile
[INFO] |     |  +- org.codehaus.plexus:plexus-component-annotations:jar:1.5.5:compile
[INFO] |     |  \- org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3:compile
[INFO] |     |     \- org.sonatype.plexus:plexus-cipher:jar:1.4:compile
[INFO] |     +- org.codehaus.plexus:plexus-utils:jar:3.0.22:compile
[INFO] |     +- org.apache.maven.shared:file-management:jar:1.2.1:compile
[INFO] |     |  +- org.apache.maven.shared:maven-shared-io:jar:1.1:compile
[INFO] |     |  |  +- org.apache.maven:maven-artifact-manager:jar:2.0.2:compile
[INFO] |     |  |  \- org.apache.maven.wagon:wagon-provider-api:jar:1.0-alpha-6:compile
[INFO] |     |  \- org.codehaus.plexus:plexus-container-default:jar:1.0-alpha-9:compile
[INFO] |     |     +- junit:junit:jar:4.13.2:compile
[INFO] |     |     |  \- org.hamcrest:hamcrest-core:jar:2.2:compile
[INFO] |     |     \- classworlds:classworlds:jar:1.1-alpha-2:compile
[INFO] |     +- org.apache.maven.reporting:maven-reporting-api:jar:3.0:compile
[INFO] |     |  \- org.apache.maven.doxia:doxia-sink-api:jar:1.0:compile
[INFO] |     +- org.apache.maven.reporting:maven-reporting-impl:jar:2.1:compile
[INFO] |     |  +- org.apache.maven:maven-project:jar:2.0.10:compile
[INFO] |     |  |  +- org.apache.maven:maven-profile:jar:2.0.10:compile
[INFO] |     |  |  \- org.apache.maven:maven-plugin-registry:jar:2.0.10:compile
[INFO] |     |  +- org.apache.maven.doxia:doxia-core:jar:1.1.2:compile
[INFO] |     |  |  +- org.apache.maven.doxia:doxia-logging-api:jar:1.1.2:compile
[INFO] |     |  |  +- xerces:xercesImpl:jar:2.8.1:compile
[INFO] |     |  |  +- commons-lang:commons-lang:jar:2.4:compile
[INFO] |     |  |  \- commons-httpclient:commons-httpclient:jar:3.1:compile
[INFO] |     |  +- org.apache.maven.doxia:doxia-site-renderer:jar:1.1.2:compile
[INFO] |     |  |  +- org.apache.maven.doxia:doxia-decoration-model:jar:1.1.2:compile
[INFO] |     |  |  +- org.apache.maven.doxia:doxia-module-xhtml:jar:1.1.2:compile
[INFO] |     |  |  +- org.apache.maven.doxia:doxia-module-fml:jar:1.1.2:compile
[INFO] |     |  |  +- org.codehaus.plexus:plexus-i18n:jar:1.0-beta-7:compile
[INFO] |     |  |  +- org.codehaus.plexus:plexus-velocity:jar:1.1.7:compile
[INFO] |     |  |  +- org.apache.velocity:velocity:jar:1.5:compile
[INFO] |     |  |  \- commons-collections:commons-collections:jar:3.2:compile
[INFO] |     |  \- commons-validator:commons-validator:jar:1.2.0:compile
[INFO] |     |     +- commons-beanutils:commons-beanutils:jar:1.7.0:compile
[INFO] |     |     +- commons-digester:commons-digester:jar:1.6:compile
[INFO] |     |     +- commons-logging:commons-logging:jar:1.0.4:compile
[INFO] |     |     +- oro:oro:jar:2.0.8:compile
[INFO] |     |     \- xml-apis:xml-apis:jar:1.0.b2:compile
[INFO] |     +- org.jacoco:org.jacoco.agent:jar:runtime:0.8.6:compile
[INFO] |     +- org.jacoco:org.jacoco.core:jar:0.8.6:compile
[INFO] |     |  +- org.ow2.asm:asm:jar:8.0.1:compile
[INFO] |     |  +- org.ow2.asm:asm-commons:jar:8.0.1:compile
[INFO] |     |  |  \- org.ow2.asm:asm-analysis:jar:8.0.1:compile
[INFO] |     |  \- org.ow2.asm:asm-tree:jar:8.0.1:compile
[INFO] |     \- org.jacoco:org.jacoco.report:jar:0.8.6:compile
```

This is a work-around users can add to their own POMs:

```
            <dependency>
                <groupId>cn.danielw</groupId>
                <artifactId>fast-object-pool</artifactId>
                <version>2.2.0</version>
                <exclusions>
                    <exclusion>
                        <groupId>org.jacoco</groupId>
                        <artifactId>jacoco-maven-plugin</artifactId>
                    </exclusion>
                </exclusions>
            </dependency>
```

**Comment by DanielYWoo** [comment]
my fault, removed and will release today.

**Comment by DanielYWoo** [comment]
will be fixed in 2.2.1

**Comment by DanielYWoo** [comment]
v2.2.1 has been released. thanks for the finding @shabino
